cli/cmd/main.go

```diff
@@ -705,13 +792,75 @@ func main() {
 	}
 }
 
+func validateDataType(c *cli.Context) (string, error) {
```

Comment:
[supernit] I think since we're already switching between data types below, it makes sense to keep the validation directly inline like you previously had and return the "type must be binary or tabular, got:" in the default case. A few reasons:

1. I think when the logic is simple enough, it's preferable to keep validations together with use cases for increased readability. 
This differs from our filter validation in app where there's various combinations of invalid filters and more complicated query inference/population, but that's already created some issues of "how do I know in this function that I can access index 0? oh, I checked than len(arr) > 0 in the validate function"
2. As we add more API calls, they may be specific to binary (e.g. add/delete tags to existing images)

Reply:
updated